### PR TITLE
Fix/523 unable to get graphql schema

### PR DESCRIPTION
**Pull request by davidverdu**
## Description

This is a bug fix that solves the problem that occurs when user deploys a project without ReadModels or without Commands. In those cases the generated GraphQL schema contained validation errors and the Lambda crashed. Now the system behaves this way:
- when there is no commands, the system returns no mutations
- When there is no read model, the system returns a _dummy query. It is mandatory for a GraphQL schema to always contain a query

Fixes: #523 

## Changes
- I updated the framework-core to take into account these edge cases
- I updated framework-core unit tests

## Checks
- [x] Project Builds
- [x] Project passes tests and checks
- [ ] Updated documentation accordingly

### CODE REVIEW

File: packages/framework-core/src/services/graphql/graphql-query-generator.ts [hunk 1]
```diff
@@ -37,12 +38,18 @@ export class GraphQLQueryGenerator {
...
     const filterQueries = this.generateFilterQueries()
+    const fields = {...byIDQueries, ...filterQueries}
+    if (Object.keys(fields).length === 0) {
+      return new GraphQLObjectType({
+        name: 'Query',
+        fields: {
+          _dummy: { type: GraphQLString }
+        }
+      })
+    }
```
Comment (davidverdu) [inline packages/framework-core/src/services/graphql/graphql-query-generator.ts:49]: It is mandatory to always return a query, even though is a silly query. Empty queries violates GraphQL validations

File: packages/framework-core/src/services/graphql/graphql-generator.ts [hunk 2]
```diff
@@ -55,7 +55,7 @@ export class GraphQLGenerator {
       this.queryGenerator,
       this.subscriptionByIDResolverBuilder.bind(this),
       this.subscriptionResolverBuilder.bind(this)
-    )
+    )
```
Comment (NickSeagull) [inline packages/framework-core/src/services/graphql/graphql-generator.ts:58]: Note these spaces, remember to always run `check-all-the-things` before submitting the PR so the linter fixes these things

File: packages/framework-core/src/services/graphql/graphql-mutation-generator.ts [hunk 3]
```diff
@@ -9,8 +9,11 @@ export class GraphQLMutationGenerator {
     private readonly mutationResolver: ResolverBuilder
   ) {}
 
-  public generate(): GraphQLObjectType {
+  public generate(): GraphQLObjectType | undefined {
```
Comment (NickSeagull) [inline packages/framework-core/src/services/graphql/graphql-mutation-generator.ts:12]: Why do we have to return `undefined`?

File: packages/framework-core/src/services/graphql/graphql-query-generator.ts (hunk 1 above)
Comment (NickSeagull) [reply packages/framework-core/src/services/graphql/graphql-query-generator.ts:49]: Perhaps we can give it a meaningful name? Like `NoQueriesDefined`?

File: packages/framework-core/src/services/graphql/graphql-generator.ts (hunk 2 above)
Comment (davidverdu) [reply packages/framework-core/src/services/graphql/graphql-generator.ts:58]: OK I'll check it

File: packages/framework-core/src/services/graphql/graphql-query-generator.ts (hunk 1 above)
Comment (davidverdu) [reply packages/framework-core/src/services/graphql/graphql-query-generator.ts:49]: I'll change its name accordingly

File: packages/framework-core/src/services/graphql/graphql-mutation-generator.ts (hunk 3 above)
Comment (davidverdu) [reply packages/framework-core/src/services/graphql/graphql-mutation-generator.ts:12]: If we return a GraphQLObjectType with name 'Mutation' with empty fields, GraphQL library throws a validation error. By sending undefined to the GraphQLGenerator it creates the schema without mutations and without triggering a validation error.

**Comment by davidverdu** [comment]
@NickSeagull I applied your suggested changes. Could you check it again?

**Review by alvaroloes — APPROVED** [review]

**Review by NickSeagull — APPROVED** [review]
Thank you a lot! 🙏
